core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -102,19 +94,21 @@
 
   private synchronized void appendCurrentBuffer(byte[] buf, int offset, int length)
       throws IOException {
-    if (!mTachyonFS.requestSpace(length)) {
-      mCanWrite = false;
-
-      String msg =
-          "Local tachyon worker does not have enough " + "space (" + length + ") or no worker for "
-              + mFile.mFileId + " " + mBlockId;
-
-      throw new IOException(msg);
+    if (mAvailableBytes < length) {
```

Comment:
Looks Like available bytes is trying to buffer ... And the code already buffers.  So why not only have one buffer and remove available bytes? I bring this up because the current Master is trying to do what you are doing and this is a current bug that causes the data registered to not match what gets written.